sycl/include/sycl/ext/oneapi/group_algorithm.hpp

```diff
@@ -65,6 +65,125 @@ using EnableIfIsNonNativeOp = cl::sycl::detail::enable_if_t<
         !cl::sycl::detail::is_native_op<T, BinaryOperation>::value,
     T>;
 
+namespace detail {
+template <typename Group> constexpr auto group_to_scope() {
+  if constexpr (std::is_same<Group, sycl::ext::oneapi::sub_group>::value) {
+    return __spv::Scope::Subgroup;
+  } else {
+    return __spv::Scope::Workgroup;
+  }
+}
+} // namespace detail
+
+/// Asynchronously copies a number of elements specified by \p numElements
+/// from the source pointed by \p src to destination pointed by \p dest
+/// with a source stride specified by \p srcStride, and returns a SYCL
+/// device_event which can be used to wait on the completion of the copy.
+/// Permitted types for dataT are all scalar and vector types, except boolean.
+template <typename Group, typename dataT>
+detail::enable_if_t<is_group_v<Group> && !detail::is_bool<dataT>::value,
+                    device_event>
```

Comment:
This is C++14 feature, we can use it safely
```suggestion
std::enable_if_t<is_group_v<Group> && !detail::is_bool<dataT>::value,
                    device_event>
```

Reply:
done